src/tests/backend/specs/enforceLowerCasePadIds.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const assert = require('assert').strict;
+const common = require('../common');
+const padManager = require('../../../node/db/PadManager');
+const settings = require('../../../node/utils/Settings');
+
+describe(__filename, function () {
```

Comment:
More tests that I would like to see added:

  * Users should be unable to manually connect via socket.io and create a pad with upper-case letters. If they can, then a malicious user would be able to hijack a URL containing upper-case letters. (For example: User A creates a pad at `https://example.com/p/foobar` but shares `https://example.com/p/FooBar` in an announcement because it's easier to read. User B uses socket.io to create a pad named `FooBar` with malicious content, thus hijacking users that use the `FooBar` spelling.
  * Users should be able to connect (via socket.io) to an existing pad with upper-case letters.

See:

  * [`common.connect()`](https://github.com/ether/etherpad-lite/blob/074482afea74d7788228fa50fca0662e4d9962fe/src/tests/backend/common.js#L132-L167)
  * [`common.handshake()`](https://github.com/ether/etherpad-lite/blob/074482afea74d7788228fa50fca0662e4d9962fe/src/tests/backend/common.js#L169-L189)
  * [example use](https://github.com/ether/etherpad-lite/blob/074482afea74d7788228fa50fca0662e4d9962fe/src/tests/backend/specs/socketio.js#L60-L62)
  * [`securityManager.checkAccess()`](https://github.com/ether/etherpad-lite/blob/074482afea74d7788228fa50fca0662e4d9962fe/src/node/db/SecurityManager.js#L53)

Reply:
> Users should be unable to manually connect via socket.io and create a pad with upper-case letters. If they can, then a malicious user would be able to hijack a URL containing upper-case letters. (For example: User A creates a pad at https://example.com/p/foobar but shares https://example.com/p/FooBar in an announcement because it's easier to read. User B uses socket.io to create a pad named FooBar with malicious content, thus hijacking users that use the FooBar spelling.

That's a new feature I suppose I should implement?

Reply:
The question being: Why is the socket padId not being santized?

Reply:
hi! Just a friendly reminder @rhansen

Reply:
Apologies for taking so long to get back to you, I got caught up with other stuff.

> The question being: Why is the socket padId not being santized?

Under normal circumstances, the pad ID sent over websocket is the pad ID seen by the browser, which is the sanitized ID (because the browser is redirected to the sanitized ID before it connects over websocket). However, due to an oversight, the server doesn't enforce sanitized IDs—it assumes the client behaves properly, which is dangerous programming practice.

Without this feature, it doesn't really matter whether the server enforces sanitized IDs because it's relatively difficult for a malicious client to use the permissiveness to its advantage. This feature changes the balance—the missing check means it is now easy for a malicious client to hijack any URL with a capital letter.

Reply:
hi @rhansen I finally got time to finish this PR. I implemented the behaviour in https://github.com/ether/etherpad-lite/pull/5501/commits/8ceb66f3eab0e0ff7a8f2d74c49da7d28bd6575e